program_intent_engagement/apps/api/v1/tests/test_views.py

```diff
@@ -0,0 +1,150 @@
+
+        self.program_id1 = str(uuid.uuid4())
+        self.program_id2 = str(uuid.uuid4())
+
+        self.url = reverse('api:v1:intents-list')
+
+    def get_api(self, user):
+        """
+        Helper function to make a get request to the API
+        """
+
+        headers = self.build_jwt_headers(user)
+        return self.client.get(self.url, **headers)
+
+    def get_response(self, user, expected_response):
+        """
+        Helper function to get API response
+        """
+
+        response = self.get_api(user)
+        self.assertEqual(response.status_code, expected_response)
+
+        return response
+
+    def test_certain_and_recent_priority(self):
+        """
+        Test that even when there's a more recent maybe it grabs the most recent certain
+        """
+        ProgramIntent.objects.create(
+            user = self.user,
+            program_uuid = self.program_id1,
+            reason = 'a reason',
+            certainty = 'CERTAIN_YES',
+            effective_timestamp = '2021-07-01 00:00:00'
+        )
+        ProgramIntent.objects.create(
+            user = self.user,
+            program_uuid = self.program_id1,
+            reason = 'a reason',
+            certainty = 'MAYBE',
+            effective_timestamp = '2021-08-01 00:00:00'
+        )
+
+        response = self.get_response(self.user, 200)
+
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]['certainty'], 'CERTAIN_YES')
+
+        return response
+
+    def test_maybe_request_taken_if_no_certain(self):
+        """
+        Test that when user has no certain program intents it returns the maybe
+        """
+
+        ProgramIntent.objects.create(
+            user = self.user,
+            program_uuid = self.program_id2,
+            reason = 'a reason',
+            certainty = 'MAYBE',
+            effective_timestamp = '2021-07-01 00:00:00'
+        )
+
+        response = self.get_response(self.user, 200)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]['certainty'], 'MAYBE')
+
+    def test_user_no_intents(self):
+        """
+        Test that when user has no program intents you get 404 response
+        """
+        new_user = UserFactory()
+        self.get_response(new_user, 404)
+
+
+    def test_multiple_intents(self):
+        """
+        Test that when user has multiple program intents you get the correct responses
+        """
+        expected_intent1 = ProgramIntent.objects.create(
+            user = self.user,
+            program_uuid = self.program_id1,
+            reason = 'a reason',
+            certainty = 'CERTAIN_YES',
+            effective_timestamp = '2021-07-01 00:00:00'
+        )
+        ProgramIntent.objects.create(
+            user = self.user,
+            program_uuid = self.program_id1,
+            reason = 'a reason',
+            certainty = 'MAYBE',
+            effective_timestamp = '2021-08-01 00:00:00'
+        )
```

Comment:
this seems the same as the maybe/certain test above

Reply:
it is 😓. Since Michael also mentioned this let's merge this convo [here](https://github.com/edx/program-intent-engagement/pull/12#discussion_r919132808)